utilities/pipelines/resourcePublish/Get-ModulesToUpdate.ps1

```diff
@@ -0,0 +1,373 @@
+﻿#region Helper functions
+
+<#
+.SYNOPSIS
+Get modified files between two commits.
+
+.PARAMETER Commit
+A git reference to base the comparison on.
+
+.PARAMETER CompareCommit
+A git reference to compare with.
+
+.EXAMPLE
+Get-ModifiedFiles -Commit "HEAD^" -CompareCommit "HEAD"
+
+    Directory: C:\Repo\Azure\ResourceModules\utilities\pipelines\resourcePublish
+
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+la---          08.12.2021    15:50           7133 Script.ps1
+
+Get modified files between previous and current commit.
+#>
+function Get-ModifiedFiles {
+    # BUG: When developing, the current logic will require you to modify the deploy.bicep/json file for it to register the change.
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false)]
+        [string] $Commit = 'HEAD',
+
+        [Parameter(Mandatory = $false)]
+        [string] $CompareCommit = 'HEAD^'
+    )
+
+    Write-Verbose "Gathering modified files between [$Commit] and [$CompareCommit]"
+    $Diff = git diff --name-only --diff-filter=AM $Commit $CompareCommit
+    $ModifiedFiles = $Diff | Get-Item
+    Write-Verbose 'The following files have been modified:'
+    $ModifiedFiles | ForEach-Object {
+        Write-Verbose (' - [{0}]' -f $_.FullName)
+    }
```

Comment:
Use the `Verbose` common parameter, to display any verbose message, regardless of the value of the $VerbosePreference variable. Here and in all Write-Verbose below

```suggestion
    Write-Verbose "Gathering modified files between [$Commit] and [$CompareCommit]" -Verbose
    $Diff = git diff --name-only --diff-filter=AM $Commit $CompareCommit
    $ModifiedFiles = $Diff | Get-Item
    Write-Verbose 'The following files have been modified:' -Verbose
    $ModifiedFiles | ForEach-Object {
        Write-Verbose (' - [{0}]' -f $_.FullName) -Verbose
    }
```